internal/data/query_utils.go

```diff
@@ -35,3 +36,92 @@ func PrepareNamedQuery(ctx context.Context, connectionPool db.ConnectionPool, na
 
 	return query, args, nil
 }
+
+// PaginatedQueryConfig contains configuration for building paginated queries
+type paginatedQueryConfig struct {
+	// Base table configuration
+	TableName    string // e.g., "operations" or "transactions"
+	CursorColumn string // e.g., "id" or "to_id"
+
+	// Join configuration
+	JoinTable     string // e.g., "operations_accounts" or "transactions_accounts"
+	JoinCondition string // e.g., "operations_accounts.operation_id = operations.id"
+
+	// Query parameters
+	Columns        string
+	AccountAddress string
+	Limit          *int32
+	Cursor         *int64
+	Forward        bool
+}
+
+// BuildPaginatedQuery constructs a paginated SQL query with cursor-based pagination
+func buildGetByAccountAddressQuery(config paginatedQueryConfig) (string, []any) {
+	var queryBuilder strings.Builder
+	var args []any
+	argIndex := 1
+
+	// Base query with join
+	queryBuilder.WriteString(fmt.Sprintf(`
+		SELECT %s, %s.%s as cursor
+		FROM %s
+		INNER JOIN %s 
+		ON %s 
+		WHERE %s.account_id = $%d`,
+		config.Columns,
+		config.TableName,
+		config.CursorColumn,
+		config.TableName,
+		config.JoinTable,
+		config.JoinCondition,
+		config.JoinTable,
+		argIndex))
+	args = append(args, config.AccountAddress)
+	argIndex++
+
+	// Add cursor condition if provided
+	if config.Cursor != nil {
+		if config.Forward {
+			queryBuilder.WriteString(fmt.Sprintf(` AND %s.%s < $%d`, config.TableName, config.CursorColumn, argIndex))
+		} else {
+			queryBuilder.WriteString(fmt.Sprintf(` AND %s.%s > $%d`, config.TableName, config.CursorColumn, argIndex))
+		}
+		args = append(args, *config.Cursor)
+		argIndex++
+	}
+
+	// Add ordering
+	// Forward: newer to older (DESC), Backward: older to newer (ASC)
+	if config.Forward {
+		queryBuilder.WriteString(fmt.Sprintf(" ORDER BY %s.%s DESC", config.TableName, config.CursorColumn))
+	} else {
+		queryBuilder.WriteString(fmt.Sprintf(" ORDER BY %s.%s ASC", config.TableName, config.CursorColumn))
+	}
```

Comment:
Why not use `Order`? `Forward` seems to me like a new concept you have to wrap your head around.

Reply:
Changed it to `isDescending` in https://github.com/stellar/wallet-backend/pull/279/commits/bbbb2ddd3afca517268e786b01ac05098109513f